### PR TITLE
Implementing pytest new strategy 

### PR DESCRIPTION
**Pull request by RaulRPrado**
Following up on #195, I implemented all the pytest.mark.skipif in all test modules.

### CODE REVIEW

File: simtools/tests/test_array_model.py [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
+    has_config_file,
+    DB_CONNECTION_MSG,
+    CONFIG_FILE_MSG,
+)
 
 logger = logging.getLogger()
 logger.setLevel(logging.DEBUG)
 
 
+@pytest.mark.skipif(not has_config_file(), reason=CONFIG_FILE_MSG)
```
Comment (GernotMaier) [inline simtools/tests/test_array_model.py:18]: I guess there is nothing significantly different between North and South that we want to run the tests for both sites.

File: simtools/tests/test_ray_tracing.py [hunk 2]
```diff
@@ -1,21 +1,27 @@
 #!/usr/bin/python3
 
+import pytest
```
Comment (GernotMaier) [inline simtools/tests/test_ray_tracing.py:3]: Suprised that the linter allows you to break alphabetical order of imports.

**Review by GernotMaier — APPROVED** [review]
All looks good (two very minor comments).

Very elegant solution, but I think we should on the medium term consider making it obsolete: tests are most useful of they are most complete and running in the background.

**Comment by GernotMaier** [comment]
Nice to see the messages now: 

simtools/tests/test_db_handler.py::TestDBHandler::test_reading_db_mst_fc SKIPPED (Connection with the DB is not available)

**Comment by GernotMaier** [comment]
Was too fast, not all tests pass for me. I did this on a completely new clone / checkout of this branch without modify any configuration parameters just doing `pytest`. See attachment.
[gm_pytest_output.log](https://github.com/gammasim/gammasim-tools/files/8117492/gm_pytest_output.log)

File: simtools/tests/test_array_model.py (hunk 1 above)
Comment (RaulRPrado) [reply simtools/tests/test_array_model.py:18]: No, we don't. For the sake of unit tests, the specific site is not relevant.

**Comment by RaulRPrado** [comment]
> simtools/tests/test_simtel_histograms.py::test_histograms takes notable longer than all other tests. Don't really see why, is it downloading that test file? run2_gamma_za20deg_azm0deg-North-Prod5_test-production-5.hdata.zst

I also noticed that.
It is probably because we have to read a large number of simtel histograms. I will try to make this test faster by selecting a subset of the histograms.

**Comment by RaulRPrado** [comment]
> Do we want to switch on also the pytest part in the CI?
> 
> https://github.com/gammasim/gammasim-tools/blob/master/.github/workflows/CI-tests.yml

Yes. Done.

**Comment by RaulRPrado** [comment]
> Was too fast, not all tests pass for me. I did this on a completely new clone / checkout of this branch without modify any configuration parameters just doing `pytest`. See attachment. [gm_pytest_output.log](https://github.com/gammasim/gammasim-tools/files/8117492/gm_pytest_output.log)

Did you run it twice?

In case there is no config.yml file, we have to create a dummy one to execute the tests. If this file is not removed after the session and you try to run the tests again, these errors will show up.
This is an issue. I will make the changes to remove the dummy config file after each session.

**Comment by RaulRPrado** [comment]
It should be working now, @GernotMaier. You can run it multiple times, just be sure to remove any config.yml that were left from the previous tests.
If it works, please merge it.

**Comment by GernotMaier** [comment]
All tests passed with sim_telarray installed and without DB - very nice.

Only issue is the warning

```
simtools/tests/test_layout_array.py: 42 warnings
simtools/tests/test_telescope_position.py: 2 warnings
  /workdir/external/gammasim-tools/simtools/layout/telescope_position.py:358: DeprecationWarning: This function is deprecated. See: https://pyproj4.github.io/pyproj/stable/gotchas.html#upgrading-to-pyproj-2-from-pyproj-1
    self._latitude, self._longitude = pyproj.transform(
```

It is probably me who should look after that. A quick search shows that I don't understand what is changing...

Also nice to see the CI in the github actions running.

**Review by GernotMaier — APPROVED** [review]

**Pull request merged**